From d5c5acc9fbbe2da010a867f10ebf68d1813ac168 Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Tue, 15 Oct 2013 15:09:15 +0200
Subject: [PATCH] Fix scrolling menu not working properly; Bug #701077

The fix for #695120 introduced a new bug, where long menus didn't properly get
scrolling, as it removed not only the call to gtk_widget_size_allocate() but
also gtk_widget_queue_resize().

Signed-off-by: Olivier Brunel <jjk@jjacky.com>
---
 gtk/gtkmenu.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gtk/gtkmenu.c b/gtk/gtkmenu.c
index f6b0c7a..ecfdb0f 100644
--- a/gtk/gtkmenu.c
+++ b/gtk/gtkmenu.c
@@ -1707,6 +1707,9 @@ gtk_menu_popup_for_device (GtkMenu             *menu,
    */
   gtk_menu_position (menu, TRUE);
 
+  /* Ensure scrolling is done properly */
+  gtk_widget_queue_resize (GTK_WIDGET (menu));
+
   gtk_menu_scroll_to (menu, priv->scroll_offset);
 
   /* if no item is selected, select the first one */
-- 
1.8.4.2

